r/configure

```diff
@@ -83,7 +83,10 @@ UNAME=`uname -s`
 # If in development mode, run the codegen script to render arrowExports.*
 if [ "$ARROW_R_DEV" = "true" ] && [ -f "data-raw/codegen.R" ]; then
   echo "*** Generating code with data-raw/codegen.R"
-  ${R_HOME}/bin/Rscript data-raw/codegen.R
+  if ! ${R_HOME}/bin/Rscript data-raw/codegen.R; then
+    echo "*** ERROR: codegen.R failed"
+    exit 1
+  fi
```

Comment:
I was surprised codegen failing didn't cause the CI job to fail, but it'll just use the existing (possibly outdated) files, so now at least we should catch it.